pulp_rpm/app/serializers.py

```diff
@@ -214,7 +214,7 @@ def validate(self, data):
         return data
 
     class Meta:
-        fields = tuple(set(SingleArtifactContentSerializer.Meta.fields)) + (
+        fields = tuple(set(SingleArtifactContentUploadSerializer.Meta.fields)) + (
             'name', 'epoch', 'version', 'release', 'arch', 'pkgId', 'checksum_type',
```

Comment:
I believe, all those metadata fields should be marked read_only on the serializer, and their values should be calculated from the artifact in `deferred_validate`.